packages/orchestrator/internal/sandbox/uffd/cross_process_helper_test.go

```diff
@@ -0,0 +1,360 @@
+package uffd
+
+// This tests is creating uffd in the main process and handling the page faults in another process.
+// It prevents problems with Go mmap during testing (https://pojntfx.github.io/networked-linux-memsync/main.html#limitations) and also more accurately simulates what we do with Firecracker.
+// These problems are not affecting Firecracker, because:
+// 1. It is a different process that handles the page faults
+// 2. Does not use garbage collection
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"golang.org/x/sys/unix"
+
+	"github.com/e2b-dev/infra/packages/orchestrator/internal/sandbox/uffd/fdexit"
+	"github.com/e2b-dev/infra/packages/orchestrator/internal/sandbox/uffd/mapping"
+	"github.com/e2b-dev/infra/packages/orchestrator/internal/sandbox/uffd/testutils"
+	"github.com/e2b-dev/infra/packages/orchestrator/internal/sandbox/uffd/userfaultfd"
+)
+
+// Main process, FC in our case
+func configureCrossProcessTest(t *testing.T, tt testConfig) (*testHandler, error) {
+	t.Helper()
+
+	data := testutils.RandomPages(tt.pagesize, tt.numberOfPages)
+
+	size, err := data.Size()
+	require.NoError(t, err)
+
+	memoryArea, memoryStart, unmap, err := testutils.NewPageMmap(uint64(size), tt.pagesize)
```

Comment:
Since this is only used in tests, you can clean this up by passing in `t` into `NewPageMmap` and doing err checking and cleanup inside instead of outside